crates/e2e/tests/e2e/helpers/services.rs

```diff
@@ -0,0 +1,244 @@
+    pub fn start_old_driver(&self, private_key: &[u8; 32], extra_args: Vec<String>) {
+        let args = [
+            "solver".to_string(),
+            format!("--solver-account={}", hex::encode(private_key)),
+            "--settle-interval=1".to_string(),
+            format!("--transaction-submission-nodes={NODE_HOST}"),
+        ]
+        .into_iter()
+        .chain(self.api_autopilot_solver_arguments())
+        .chain(extra_args.into_iter());
+
+        let args = solver::arguments::Arguments::try_parse_from(args).unwrap();
+        tokio::task::spawn(solver::run::run(args));
+    }
+
+    async fn wait_for_api_to_come_up() {
+        let is_up = || async {
+            reqwest::get(format!("{API_HOST}{VERSION_ENDPOINT}"))
+                .await
+                .is_ok()
+        };
+
+        tracing::info!("Waiting for API to come up.");
+        wait_for_condition(Duration::from_secs(10), is_up)
+            .await
+            .expect("waiting for API timed out");
+    }
+
+    pub async fn get_auction(&self) -> AuctionWithId {
+        let response = self
+            .http
+            .get(format!("{API_HOST}{AUCTION_ENDPOINT}"))
+            .send()
+            .await
+            .unwrap();
+
+        let status = response.status();
+        let body = response.text().await.unwrap();
+
+        assert_eq!(status, StatusCode::OK, "{body}");
+
+        serde_json::from_str(&body).unwrap()
+    }
+
+    /// Create an [Order].
+    /// If the response status code is not `201`, return the status and the
+    /// body.
+    pub async fn create_order(
+        &self,
+        order: &OrderCreation,
+    ) -> Result<OrderUid, (StatusCode, String)> {
+        let placement = self
+            .http
+            .post(format!("{API_HOST}{ORDERS_ENDPOINT}"))
+            .json(order)
+            .send()
+            .await
+            .unwrap();
+
+        let status = placement.status();
+        let body = placement.text().await.unwrap();
+
+        match status {
+            StatusCode::CREATED => Ok(serde_json::from_str(&body).unwrap()),
+            code => Err((code, body)),
+        }
+    }
+
+    /// Submit an [model::quote::OrderQuote].
+    /// If the response status is not `200`, return the status and the body.
+    pub async fn submit_quote(
+        &self,
+        quote: &OrderQuoteRequest,
+    ) -> Result<OrderQuoteResponse, (StatusCode, String)> {
+        let quoting = self
+            .http
+            .post(&format!("{API_HOST}{QUOTING_ENDPOINT}"))
+            .json(&quote)
+            .send()
+            .await
+            .unwrap();
+
+        let status = quoting.status();
+        let body = quoting.text().await.unwrap();
+
+        match status {
+            StatusCode::OK => Ok(serde_json::from_str(&body).unwrap()),
+            code => Err((code, body)),
+        }
+    }
+
+    pub async fn solvable_orders(&self) -> usize {
+        self.get_auction().await.auction.orders.len()
+    }
```

Comment:
I think it would be a bit nicer to return the orders themselves instead of just the count. My reasoning is that manually calling `.len()` is not a lot of overhead and wouldn't require another function in case you want to assert something about the contents of the orders.

Reply:
`solvable_orders` was a standalone function returning the size of the current auction. There is another method `get_auction`, which returns the current auction itself. I thought about removing this entirely and using `services.get_auction().await.auction.orders.len()` everywhere, but in many places we're interested only with the count of the orders. So I left it this way :p